Update zindi package

Tim-Abwao opened the pull request:
An attempt to help improve the package by applying some guidelines and best practices I've been learning.

Some of the changes involved renaming or deleting some items. Please let me know if this is okay, and if the changes are useful.

Tim-Abwao committed:
Update setup.py

Update setup information.
Update .gitignore to avoid tracking built files.
Update license to 'The CIA TEAM'.

Tim-Abwao committed:
Update user.py

Apply some PEP8 code style guidelines.
Limit repetition of code by refactoring and leveraging object attributes.
Fix typos.

Tim-Abwao committed:
Update README.md

Tim-Abwao committed:
Update .gitignore

Avoid tracking downloaded datasets

Tim-Abwao committed:
Move docs folder out of zindi package

eaedk commented:
Hi team, thanks a lot for all your help.
I'm busy now, I'll be in touch asap.

---
Best regards / Cordialement
Mr. Emmanuel KOUPOH
Data, ML & DL Scientist - Python & Data Science Instructor
+221 78 528 25 60 || Linkedin
<https://www.linkedin.com/in/esa%C3%AFe-alain-emmanuel-dina-koupoh-7b974a17a/>


Le mar. 30 mars 2021 à 11:04, Tim Abwao ***@***.***> a écrit :

> An attempt to help improve the package by applying some guidelines and
> best practices I've been learning.
>
> Some of the changes involved renaming or deleting some items. Please let
> me know if this is okay, and if the changes are useful.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/eaedk/testing-zindi-package/pull/4
> Commit Summary
>
>    - Update setup.py
>    - Update user.py
>    - Update README.md
>    - Update .gitignore
>    - Move docs folder out of zindi package
>
> File Changes
>
>    - *M* .gitignore
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
>    (11)
>    - *M* LICENSE
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-c693279643b8cd5d248172d9c22cb7cf4ed163a3c98c8a3f69c2717edd3eacb7>
>    (2)
>    - *M* README.md
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>    (59)
>    - *R* docs/features.txt
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-7387d81c66914a8fede94b05dbad41f16ec9061d2fc16b10612d365c603689ef>
>    (0)
>    - *R* docs/template.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-233d03d8da981157320d8a3e2cec5a8da786c9ee4e7987d23423aac4493b7a50>
>    (0)
>    - *R* docs/utils/challenge_idx_selector.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-9d8997c935d04f80bbf1a25d2c645daf56ee3561cd844923ca47ec853927beb7>
>    (0)
>    - *R* docs/utils/download.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-32f71ea3ebd30193005d340dd2ff4a25e652438e2ed06e5a71543e399904041b>
>    (0)
>    - *R* docs/utils/get_challenges.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-63c85c1371d555a789bdb86f46dd59b4b8870a69814065e82251915cf6a1daf1>
>    (0)
>    - *R* docs/utils/join_challenge.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-bfa8a433f485b2432af338caafa88bae599c0527ed03ed728f3778375e45f8af>
>    (0)
>    - *R* docs/utils/n_subimissions_per_day.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-3caadfa9471899a02fba69b01e8319a0695bd38b2d4376fbd44462e467f4b2cd>
>    (0)
>    - *R* docs/utils/participations.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-697b57292a246426d367a410e12756126f18b5ad4a33f4a3df92faa61a513c29>
>    (0)
>    - *R* docs/utils/print_challenges.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-1c62978d44e73f3ab04882557cfe459a64aa85592a4d2bda6a679cdf13d206e4>
>    (0)
>    - *R* docs/utils/print_lb.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-c93da7336a79193e091306ecb3d46ad1d02412d1d7c9c7174bd68abe24a03677>
>    (0)
>    - *R* docs/utils/print_submission_board.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-cac82311a3e3d78debf43fabc0b6fc42f1249d3de86966f71238f2c2c33aef73>
>    (0)
>    - *R* docs/utils/upload.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-1ba0bc409f3155d9dc805f3e3ee446989ee71a535a6bc71ad95633227844bcde>
>    (0)
>    - *R* docs/utils/user_on_lb.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-3cb4ce7f6b10f3c68433de5231ceba66f13c310e12f9de9ab1cf0cad20d75a9d>
>    (0)
>    - *M* setup.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7>
>    (32)
>    - *D* tests/dataset/SampleSubmission.csv
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-94385d5af5109a970852d359ccf388cab1ec94bbe522dbf4aab260262076c05b>
>    (1672001)
>    - *D* tests/dataset/VariableDefinitions.txt
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-a27300b07c1868aaf61a4e1c74e0428ec779898899b1dc0dfd35e7ab0bf4f3d9>
>    (62)
>    - *D* tests/dataset/orders.csv
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-07136e0f5c1638a29a7df934afa38be6b449b817f1bd4524cfd4f053c2779475>
>    (135304)
>    - *D* tests/dataset/vendors.csv
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-52fb43b2d68e33ea86c213ca0b01b1e41a1a0643be2a99119ebc81809451c1b8>
>    (101)
>    - *M* zindi/user.py
>    <https://github.com/eaedk/testing-zindi-package/pull/4/files#diff-969e900b35f2c90c809c03af9e73c65213f32b41640285bbdb4b14960c3be942>
>    (610)
>
> Patch Links:
>
>    - https://github.com/eaedk/testing-zindi-package/pull/4.patch
>    - https://github.com/eaedk/testing-zindi-package/pull/4.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eaedk/testing-zindi-package/pull/4>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG2G3AUCBUC2ZO5TR7F7TVTTGGV3TANCNFSM42BZEVQQ>
> .
>

Tim-Abwao committed:
Improve readerbility.

Give some methods and variables more intuitive names.
Apply some style guides from PEP8.

Tim-Abwao committed:
Update .gitignore

Avoid tracking un-named jupyter notebooks.

Tim-Abwao committed:
Update README.md

Tim-Abwao committed:
Add pyproject.toml

Tim-Abwao committed:
Add sphinx docs

Create source files to generate the docs using sphinx.

Tim-Abwao committed:
Update decorated functions

Create separate functions to perform the expensive fetching and processing work, so that the @property-decorated functions remain lightweight.
User data will be updated whenever a user selects/switches challenges or explictly calls a method that updates user data.

Tim-Abwao committed:
fix warning in file-download progress-bar

Get the ceiling of the total value for the progress-bar.
Totals that differed from the progress-bar's size (probably due to round-off-error or float representation) raised a warning.

Tim-Abwao committed:
Update requirements.txt

Tim-Abwao commented on README.md:
````diff
@@ -1,80 +1,81 @@
 
+Or even:
+
+```bash
+pip install https://github.com/eaedk/testing-zindi-package/archive/master.tar.gz
+```
````
Installing from `https://github.com/eaedk/testing-zindi-package/archive/master.tar.gz` is faster than installing from  `git+https://github.com/eaedk/testing-zindi-package.git`, since it only downloads a snapshot of the latest version, not the entire history. [See stackoverflow answer][1]

[1]: https://stackoverflow.com/a/24811490

Tim-Abwao commented on README.md:
```diff
@@ -1,80 +1,81 @@
 
-In progress 🚧 . All listed functions in **Usage** part are working, but please share issues with use and request features that you need.
-
-## Documentation
-
-We will update this table soon ... Now you can refer to the notebook and **Usage** section.
-
```
The documentation can be hosted elsewhere e.g. at <https://readthedocs.org/>. This section can be replaced with a link to the docs.

Tim-Abwao commented on README.md:
```diff
@@ -1,80 +1,81 @@
-
-# Contributers
-
-<div align='center'>
-
-| <img src='https://media-exp1.licdn.com/dms/image/C5103AQEWceAkHjUwVw/profile-displayphoto-shrink_400_400/0/1582378323644?e=1618444800&v=beta&t=dgTb6nwVBgbNzFOs8CLvoM_p2gYal4t0gStKWvfzcmk' width='100' height='100' style='border-radius:50%; margin:.8cm'> <br>Emmanuel KOUPOH                        | <img src='https://media-exp1.licdn.com/dms/image/C4D35AQGOLlRWnmK5UA/profile-framedphoto-shrink_800 ...
-|--------------------------------------|-------------------------------|----------------------------------------------|
-| [eaedk](https://github.com/eaedk) | [dric2018](https://github.com/dric2018) | [NazarioR9](https://github.com/NazarioR9)|
-| [Emmanuel on linkedin](https://www.linkedin.com/in/esaïe-alain-emmanuel-dina-koupoh-7b974a17a) | [Cedric on linkedin](https://www.linkedin.com/in/cédric-pascal-emmanuel-manouan-ba9ba1181) | [Muhamed on linkedin](https://www.linkedin.com/in/muhamed-tuo-b1b3a0162) |
-|[@eaedk😂](https://zindi.africa/users/eaedk) | [@Zeus😆](https://zindi.africa/users/I_am_Zeus_AI) |   [@Nazario😁](https://zindi.africa/users/Muhamed_Tuo)   |
```
The linked images weren't being displayed.

Tim-Abwao reviewed (commented):
*Reviewing **README.md ...***

Tim-Abwao commented on docs/source/conf.py:
```diff
@@ -0,0 +1,58 @@
+sys.path.insert(0, os.path.abspath('../../'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'zindi'
+copyright = '2021, The CIA Team'
+author = 'The CIA Team'
+
+# The full version, including alpha/beta/rc tags
+release = '0.0.1'
```
Project information and configuration for the documentation. Please see <https://www.sphinx-doc.org/en/master/usage/configuration.html> for more options.

Tim-Abwao commented on docs/Makefile:
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
```
Allows you to conveniently run [Sphinx][1] commands to build docs. Typical usage is:

```bash
cd docs
sphinx-quickstart
make html
## To apply changes made to docs source-files
make clean && make html
```
[1]: https://www.sphinx-doc.org/en/master/usage/quickstart.html

Tim-Abwao commented on setup.py:
```diff
@@ -1,31 +1,29 @@
-import setuptools, sys, subprocess
+import setuptools
 
-try:
-    output = subprocess.run([sys.executable, '-m', 'pip', 'install','-r','requirements.txt'])
-except:
-    output = subprocess.run(['pip', 'install','-r','requirements.txt'])
```
`install_requires` allows automatic handling of dependencies during installation.

[install-requires-vs-requirements-files](https://packaging.python.org/discussions/install-requires-vs-requirements/#install-requires-vs-requirements-files)

Tim-Abwao commented on pyproject.toml:
```diff
@@ -0,0 +1,6 @@
+[build-system]
```
Specifies *build tools* and *build requirements*. (<https://packaging.python.org/tutorials/packaging-projects/#creating-pyproject-toml>)

Tim-Abwao reviewed (commented):
*Reviewing packaging files ...*

Tim-Abwao commented on zindi/user.py:
```diff
@@ -1,429 +1,443 @@
-
-import sys, os
-
-## To avoid errors of importing before instalation
-currentdir = os.path.dirname(os.path.realpath(__file__))
-parentdir = os.path.dirname(currentdir)
-sys.path.append(parentdir)
```
`pip` can install in [editable mode](https://pip.pypa.io/en/stable/cli/pip_install/#editable-installs):

```bash
pip install -e .
```
You can install the package, and still keep editing it.

Tim-Abwao commented on zindi/user.py:
```diff
@@ -1,429 +1,443 @@
-        """Push submission files for the selected challenge to Zindi platform.
+    @staticmethod
+    def _get_username_or_teamtitle(data):
+        """Get the
```
```suggestion
        """Get the user's username, or team-name.
```
